packages/host/app/lib/matrix-handlers/index.ts

```diff
@@ -36,8 +35,8 @@ export type Event = Partial<IEvent> & {
 };
 
 export interface EventSendingContext {
-  setRoom: (roomId: string, roomPromise: Promise<RoomField>) => void;
-  getRoom: (roomId: string) => Promise<RoomField> | undefined;
+  setRoom: (roomId: string, room: RoomModel) => void;
+  getRoom: (roomId: string) => RoomModel | undefined;
```

Comment:
@habdelra should this be a promise? 

Based upon the resource behaviour, it should be a promise, since `await getRoom` will prevent other writes occuring to the set as it is accessed.

Reply:
If so, we should add a comment here bcos `setRoom` and `getRoom` code doesn't look asynchronous but is based upon where its called -- not sure

Reply:
Ok. Then I think the change in types here are wrong. Will revert

Reply:
managing these types are problematic particularly, if we are using .events to drive the indexing of a room. Turning `getRoom` into a Promise will mean so much handling for awaiting room and I think even if you do so the trackedness is swallowed by awaiting the room. This is back to the whole scenario as described where I literally have to call `.events` before `await getRoom`. The resource already acts asynchronously. 

In fact, I think this is probably needs rethinking into our refactor of the matrix subscriber refactor ticket -- https://linear.app/cardstack/issue/CS-6987/get-room-resource-to-register-with-matrix-event-handler. Meanwhile, I have written a note at each `await getRoom` to communicate that we need to do it to handle the asynchrony outside of the subscriber

Look at this commit for the comments I have made https://github.com/cardstack/boxel/pull/1407/commits/47bc381d02630bff4092abba9bdcdedde7f0ad48